### PR TITLE
feat(MarchingCubes): add component, demo, docs

### PR DESCRIPTION
**Pull request by andretchen0**
Port of THREE/Drei's `<MarchingCubes />` to Cientos. Adds:

* component
* demo
* docs

### CODE REVIEW

**Comment by JvanderHeide** [comment]
Would love to see this merged

**Comment by andretchen0** [comment]
> Would love to see this merged

Good to hear!

File: src/core/abstractions/MarchingCubes/MarchingCubes.vue [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+const props = withDefaults(defineProps<MarchingCubesProps>(), {
+  resolution: 28,
+  maxPolyCount: 10000,
+  enableUvs: false,
+  enableColors: false,
+})
+
+const marchingCubes = computed(() => new MarchingCubesImpl(props.resolution, null as unknown as Material, props.enableUvs, props.enableColors, props.maxPolyCount))
```
Comment (alvarosabu) [inline src/core/abstractions/MarchingCubes/MarchingCubes.vue:22]: mmm... @andretchen0 I don't think we should pass a null and cast it as a material.

Should we use a `new MeshBasicMaterial` as default or maybe enable a prop to pass a material? How does the MarchingCubes get the material from? Like this?

```
<MarchingCubes :enable-colors="true" :resolution="40" :max-poly-count="40000">
      <MarchingPlane v-if="c.hasWallX.value.value" plane-type="x" />
      <MarchingCube
        :position="[c.x.value.value, c.y.value.value, c.z.value.value]"
        :strength="c.strength.value.value"
        :subtract="c.subtract.value.value"
      />
      <MarchingCube ref="r0" color="red" />


      <TresMeshBasicMaterial v-if="c.enableColors.value.value" :vertex-colors="true" />
      <TresMeshPhongMaterial v-else specular="#111111" :shininess="30" color="#049ef4" :reflectivity="1" />
    </MarchingCubes>
```

**Review by alvarosabu — CHANGES_REQUESTED** [review]
I really like this abstraction. I just have some questions about how to use materials and some typescript stuff long the way

File: src/core/abstractions/MarchingCubes/MarchingCubes.vue (hunk 1 above)
Comment (alvarosabu) [reply src/core/abstractions/MarchingCubes/MarchingCubes.vue:22]: @andretchen0 I just answered my own question, changed the `MeshPhongMaterial` on the playground for `Toon Material` and all good. Also with the default `MeshBasicMaterial` on the constructor

Comment (andretchen0) [reply src/core/abstractions/MarchingCubes/MarchingCubes.vue:22]: Hey @alvarosabu 

> I just answered my own question

Great! Is this issue resolved?

Comment (andretchen0) [reply src/core/abstractions/MarchingCubes/MarchingCubes.vue]: After the follow-up on Discord, I've added a default material.

**Review by alvarosabu — APPROVED** [review]

**Pull request merged**